### DISCUSSION TITLE
Static lifetime for variable passed into ssl certs

### DISCUSSION THREAD

**Discussion by dsgipe** [blackbeam/mysql_async#287] [General]
I have used https://docs.rs/mysql_async/latest/mysql_async/struct.SslOpts.html to setup my call. But the path to the cert will be different depending on where we run it (locally, deployed). We would like to use a variable. But I can't figure out how to pass in a variable with static lifetime without leaking memory. Is that the expectation of the API? Is there a way to not use static lifetimes?

Thanks for any recommendations.

**Comment by blackbeam** [comment]
Hi.
You mean `SslOpts::with_root_cert_path`? You can give it `Cow::Owned`.
